### ISSUE TITLE
TestNotRunningError: expect() must be called in a test

### ISSUE THREAD

**Issue by xlc** [oven-sh/bun#6919]
### What version of Bun is running?

1.0.9+98f20170a

### What platform is your computer?

Darwin 23.0.0 arm64 arm

### What steps can reproduce the bug?

When a test getting timeout, the execution is not aborted and the tests will continue to run and throw more error.

This is a redacted log

```
...fetch-blocks.test.ts:
Timeout: test "doFetchBlocks" timed out after 10000ms
✗ fetch-blocks > doFetchBlocks [16160.57ms]

...builder.test.ts:
✓ sync > doCreateFetchBlocks [31.85ms]

...blocks.test.ts:
✓ Blocks > firstMissingBlock > startBlock [51.47ms]
✓ Blocks > firstMissingBlock > endBlock [4.53ms]

13 |     expect(await doFetchBlocks(startBlock, startBlock + 3, rpc)).toMatchSnapshot()
        ^
TestNotRunningError: expect() must be called in a test
      at .../fetch-blocks.test.ts:13:4
      at processTicksAndRejections (:61:76)
```

It is confusing to see the error `TestNotRunningError: expect() must be called in a test` and spend a few minutes trying to figure why I am calling `expect` outside a test. And then I found out the log `Timeout: test "doFetchBlocks" timed out after 10000ms`. So the error is actually the test code continue to run after it got timeout and it throws error on `expect`.

### What is the expected behavior?

It will be best to abort the test completely but if that's not possible or too complicated, it should toggle some other global flag to indicate `expect` that it is running on an aborted session and shouldn't throw errors.

**Comment by Electroid** [comment]
Is the `expect()` being called from a timer? (e.g. `setTimeout`) or from a lifecycle hook? (e.g. `beforeAll`, `afterEach`)

**Comment by xlc** [comment]
No. This is the code

```ts
describe('fetch-blocks', () => {
  it('doFetchBlocks', async () => {
    const endpoint = 'xxx'
    const rpc = new HttpProvider(endpoint)

    const startBlock = 10000000
    expect(await doFetchBlocks('polkadot', startBlock, startBlock, rpc)).toMatchSnapshot()
    expect(await doFetchBlocks('polkadot', startBlock, startBlock + 3, rpc)).toMatchSnapshot()
  }, 30000)
})
```

The second expect is the one throwing error

**Comment by seanwessmith** [comment]
same issue here when running a test. I even replaced the actual expect with `expect(true).toBe(true);` and it still thows an error `expect() must be called in a test`

```js
test("db Webhook Status add active status", async () => {
    const result = await sdk.shopify.dbSetWebhookStatus("123", "active");
    console.log("result.success", result.success);
    expect(true).toBe(true);
    // expect(result.success).toBe(true);
  }, 20000);
  ```
  
  ```shell
 result.success true
23 |   //   expect(result).toBeNull();
24 |   // });
25 |   test("db Webhook Status add active status", async () => {
26 |     const result = await sdk.shopify.dbSetWebhookStatus("123", "active");
27 |     console.log("result.success", result.success);
28 |     expect(true).toBe(true);
        ^
TestNotRunningError: expect() must be called in a test
```

**Comment by paperclover** [comment]
Fixed in 1.0.15 (we removed this error)
